crates/aptos-jwk-consensus/src/jwk_observer.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright © Aptos Foundation
+
+use anyhow::Result;
+use aptos_channels::aptos_channel;
+use aptos_logger::{debug, info};
+use aptos_types::jwks::{jwk::JWK, Issuer};
+use futures::{FutureExt, StreamExt};
+use move_core_types::account_address::AccountAddress;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use tokio::{sync::oneshot, task::JoinHandle, time::MissedTickBehavior};
+
+#[derive(Serialize, Deserialize)]
+struct OpenIDConfiguration {
+    issuer: String,
+    jwks_uri: String,
+}
+
+#[derive(Serialize, Deserialize)]
+struct JWKsResponse {
+    keys: Vec<serde_json::Value>,
+}
+
+/// Given an Open ID configuration URL, fetch its JWKs.
+pub async fn fetch_jwks(my_addr: AccountAddress, config_url: Vec<u8>) -> Result<Vec<JWK>> {
+    if cfg!(feature = "smoke-test") {
+        use reqwest::header;
+        let maybe_url = String::from_utf8(config_url);
+        let jwk_url = maybe_url?;
+        let client = reqwest::Client::new();
+        let JWKsResponse { keys } = client
+            .get(jwk_url.as_str())
+            .header(header::COOKIE, my_addr.to_hex())
```

Comment:
is the difference just the header?

Reply:
in smoke test it's 1 request; in reality it's 1 request for open_id_config, then a 2nd call for open_id_config.jwk_url